Rakefile

```diff
@@ -34,57 +34,27 @@ rescue LoadError => e
   # rspec won't exist on production
 end
 
-def read_index(uri)
-  Zlib::GzipReader.open(open(uri)) {|gz| Marshal.load(gz) }
-end
-
-def modified?(uri, cache_file)
-  uri   = URI(uri)
-  file  = nil
-
-  file = File.stat(cache_file) if File.exists?(cache_file)
+def download_index(url)
+  uri = URI(url)
 
-  req = Net::HTTP::Get.new(uri.request_uri)
-  req['If-Modified-Since'] = file.mtime.rfc2822 if file
+  Tempfile.create(uri.path) do |file|
+    file.write(open(uri.to_s).read)
+    file.rewind
 
-  res = Net::HTTP.start(uri.hostname, uri.port) {|http|
-    http.request(req)
-  }
-
-  if res.response['Location']
-    modified?(res.response['Location'], cache_file)
-  elsif res.is_a?(Net::HTTPSuccess)
-    File.open(cache_file, 'w') {|file| file.write res.body }
-    true
-  else
-    false
+    Zlib::GzipReader.open(file) do |gz|
+      Marshal.load(gz)
+    end
```

Comment:
We already run this task as a corn every 60 minutes, so retrying didn't seem super important to me.

On Wed, Jul 29, 2015 at 3:17 AM, André Medeiros <notifications@github.com>
wrote:

>>  
>> -  res = Net::HTTP.start(uri.hostname, uri.port) {|http|
>> -    http.request(req)
>> -  }
>> -
>> -  if res.response['Location']
>> -    modified?(res.response['Location'], cache_file)
>> -  elsif res.is_a?(Net::HTTPSuccess)
>> -    File.open(cache_file, 'w') {|file| file.write res.body }
>> -    true
>> -  else
>> -    false
>> +    Zlib::GzipReader.open(file) do |gz|
>> +      Marshal.load(gz)
>> +    end
> Sounds to me like we could be rescuing and retrying a couple times here maybe?
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/bundler/bundler-api/pull/95/files#r35745391